objects/src/accounts/data.rs

```diff
@@ -0,0 +1,212 @@
+use std::{
+    fs::{self, File},
+    io::{self, Read},
+    path::Path,
+};
+
+use miden_crypto::utils::SliceReader;
+
+use super::{
+    super::utils::serde::{
+        ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable,
+    },
+    Account, Word,
+};
+
+// ACCOUNT DATA
+// ================================================================================================
+
+/// AccountData is a representation of the Account struct meant to be used
+/// for Account serialisation and deserialisation for transport of Account data
+#[derive(Debug, PartialEq, Eq)]
+pub struct AccountData {
+    pub account: Account,
+    pub account_seed: Option<Word>,
+    pub auth: AuthData,
+}
+
+impl AccountData {
+    pub fn new(account: Account, account_seed: Option<Word>, auth: AuthData) -> Self {
+        Self { account, account_seed, auth }
+    }
+
+    /// Serialises and writes binary AccountData to specified file
+    pub fn write(&self, filepath: &Path) -> Result<(), io::Error> {
```

Comment:
IMO this should be generic over `AsRef<Path>`:

```suggestion
    pub fn write(&self, filepath: impl AsRef<Path>) -> Result<(), io::Error> {
```

This allows the method to be called with basically every string representation from the stdlib.

Reply:
Btw, there is a [typealias for `Result<T, io::Error>`](https://doc.rust-lang.org/std/io/type.Result.html) in the `io` module, this could be:

```suggestion
    pub fn write(&self, filepath: &Path) -> io::Result<()> {
```

Reply:
added.